packages/contract/src/__test__/coverage-contract/coverage-contract.test.ts

```diff
@@ -460,4 +460,67 @@ describe('Coverage Contract', () => {
 
     expect(receiverMessages).toStrictEqual(messages);
   });
+
+  it('should echo vector output via log', async () => {
+    const INPUT = [23, 6, 1, 51, 2];
+    const { value } = await contractInstance.functions.echo_u8_vector(INPUT).call();
+
+    expect(value).toStrictEqual(INPUT);
+  });
+
+  it('should echo struct vector output via log', async () => {
+    const INPUT = [
+      {
+        foo: 1,
+        bar: 10,
+      },
+      {
+        foo: 2,
+        bar: 20,
+      },
+      {
+        foo: 3,
+        bar: 30,
+      },
+    ];
+    const { value } = await contractInstance.functions.echo_struct_vector(INPUT).call();
+    expect(value).toStrictEqual(INPUT);
+  });
+
+  it('should echo complex struct vector output via log', async () => {
+    type ComplexStruct = {
+      foo: number;
+      bar: BigNumberish;
+      baz: string;
+    };
+    const INPUT: ComplexStruct[] = [
+      {
+        foo: 1,
+        bar: 11337,
+        baz: '123456789',
+      },
+      {
+        foo: 2,
+        bar: 21337,
+        baz: 'alphabet!',
+      },
+      {
+        foo: 3,
+        bar: 31337,
+        baz: 'dogs cats',
+      },
+      {
+        foo: 4,
+        bar: 41337,
+        baz: 'nineChars',
+      },
+    ];
+    const EXPECTED = INPUT.map((input) => ({ ...input, bar: bn(input.bar).toHex() }));
+    const { value } = await contractInstance.functions.echo_complex_struct_vector(INPUT).call();
+    const RESULT = value.map((result: ComplexStruct) => ({
+      ...result,
+      bar: bn(result.bar).toHex(),
+    }));
+    expect(RESULT).toStrictEqual(EXPECTED);
+  });
```

Comment:
We should add a test using two methods that return Vec in a multicall. The inputs would need to be different in order to test if it will correctly get each returned log.

```ts
contractInstance.multiCall([
    contractInstance.functions.echo_complex_struct_vector(INPUT),
    contractInstance.functions.echo_struct_vector(INPUT_2),
])
```